backend/manager/modules/restapi/jaxrs/src/main/java/org/ovirt/engine/api/restapi/resource/BackendDiskSnapshotsResource.java

```diff
@@ -35,16 +38,28 @@ public DiskSnapshots list() {
 
     protected DiskSnapshots mapCollection(List<org.ovirt.engine.core.common.businessentities.storage.Disk> entities) {
         DiskSnapshots collection = new DiskSnapshots();
+        Mapper<org.ovirt.engine.core.common.businessentities.storage.Disk, DiskSnapshot> mapper =
+                getMapper(org.ovirt.engine.core.common.businessentities.storage.Disk.class, DiskSnapshot.class);
+
         for (org.ovirt.engine.core.common.businessentities.storage.Disk disk : entities) {
-            DiskSnapshot diskSnapshot = getMapper(org.ovirt.engine.core.common.businessentities.storage.Disk.class, DiskSnapshot.class).map(disk, null);
+            DiskSnapshot diskSnapshot = mapper.map(disk, null);
             diskSnapshot.setDisk(new Disk());
             diskSnapshot.getDisk().setId(this.diskId.toString());
             collection.getDiskSnapshots().add(addLinks(populate(diskSnapshot, disk), Disk.class));
-            diskSnapshot.setHref(buildHref(diskId.toString(), diskSnapshot.getId().toString()));
+            diskSnapshot.setHref(buildHref(diskId.toString(), diskSnapshot.getId()));
+            getBackendCollection(QueryType.GetDiskSnapshotByImageId,
+                    new DiskSnapshotsQueryParameters(diskId, includeActive(), includeTemplate()));
             if (diskSnapshot.getParent() != null) {
-                diskSnapshot.getParent().setHref(buildParentHref(diskId.toString(), false));
+                QueryReturnValue result = runQuery(QueryType.GetDiskIdBySnapshotId,
```

Comment:
You don't need this query.
Look at the diskSnapshot object before you run `setDisk(new Disk())`

Reply:
If the include_template flag is not included I need to do that query. 
@ahadas what do you think is the best way to resolve this?
Get the parent disk in GetAllDiskSnapshots Query,  
maybe add if statement when the flag is raised or not,
or when the flag is not raised set the parent as null

Reply:
makes sense to me to set it to null in that case, as the template volumes are not included

Reply:
Ok wiil be done

Reply:
I think that the parent should not be null if there is a parent, following that logic a single GET would never have this field.
I think that the correct way is to handle it in the GetAllDiskSnapshotsQuery, and then the Backend would get a correct parent diskId and there will be no need in all of this logic here...

Reply:
Same for BackendDiskSnapshotResource

Reply:
Is there any decision on how I should proceed?

Reply:
Please look if you can handle it efficiently in GetAllDiskSnapshotsQuery and GetDiskSnapshotByImageIdQuery. If not, then let's leave the parent as null if include_template flag is not raised.